src/io/tree.cpp

```diff
@@ -517,6 +517,8 @@ Tree::Tree(const std::string& str) {
   } else {
     shrinkage_ = 1.0f;
   }
+
+  RecomputeLeafDepths();
```

Comment:
my only concern is this may slow down the model loading speed. @wxchan thoughts ?

Reply:
@guolinke I think it will slow down. Can we do a test on performance here?

I didn't read all codes of this part. Is it possible to make it optional? Like only user wants to predict contrib to run this.

Reply:
I can make it lazy but then I can't make `PredictContrib` `const`...a small issue

Reply:
I should note that I originally did it on model load since it is nice for a model to look the same after it has been loaded as after it has been trained.